src/NServiceBus.Core/Configure.cs

```diff
@@ -126,6 +127,23 @@ internal void Initialize()
             }
         }
 
+        void ConfigureBehaviors()
```

Comment:
Well, it affects structuremap users that use AssertConfigurationIsValid. There is no runtime issue due to incomplete registrations as mentioned in #3468 .

Just renaming 5.2.15 to 5.3.0 is not enough to support backward compatibility.

Again the question is, was it the intent in the first place to always register all scanned `IBehavior<>` even when not used at runtime? It this was NOT the intent, then we have a bug and this patch fixes this bug. It break behavior with previous V5 versions but that would in this perspective not matter.

@andreasohlund @timbussmann @SimonCropp opinion?